maptivist/src/test/java/com/google/sps/MarkerServletFunctionTest.java

```diff
@@ -14,66 +14,95 @@
- 
-    private MarkerServlet testMarkerServletFunctions;
- 
+    private final LocalServiceTestHelper helper =
+        new LocalServiceTestHelper(new LocalDatastoreServiceTestConfig());
+
+    // All the different mock examples of marker info obtained by servlet through POST method or sent by querystring
+    private final String TITLE = "Justice for Breonna Taylor";
+
+    private final String DESCRIPT = "We will walk to the capital";
+
+    private final double LAT = 32.565;
+    private final double LONG = 45.8574;
+
+    private final String CAT_A = "BLM";
+    private final String CAT_B = "LGBTQ";
+    private final String CAT_C = "Policy";
+
+    private final String LINK_A = "google.com";
+    private final String LINK_B = "bing.com";
+
+    private final String FLAG_A = "Hateful, message";
+    private final String FLAG_B = "Not real";
+
+    private final int VOTES = 0;
+
+    private final UUID ID = UUID.randomUUID();
+    
     @Before
     public void setUp() {
-        testMarkerServletFunctions = new MarkerServlet();
-    }
- 
-    @Test
-    public void matchingDelimiterCounts() {
-        String categoriesString = testMarkerServletFunctions.createCategoriesString(categoryObject);
-        int expected = 1;
-        long actual = categoriesString.chars().filter(delim -> delim == ',').count();
-        Assert.assertEquals(actual, expected);
- 
+        helper.setUp();
     }
- 
-    @Test
-    public void delimiterIncludedInString() {
-        String flagString = testMarkerServletFunctions.createFlagString(flagObject);
-        int expected = 2;
-        long actual = flagString.chars().filter(delim -> delim == ',').count();
 
-        Assert.assertEquals(actual, expected);
-    }
- 
-    @Test
-    public void inverseFunctionsWorkTogether() {
-        String linkString = testMarkerServletFunctions.createLinkString(linkObject);
-        Set<String> expected = testMarkerServletFunctions.createLinkObject(linkString);
- 
-        Assert.assertEquals(linkObject, expected);
+  @After
+    public void tearDown() {
+        helper.tearDown();
     }
+
+  @Test
+  public void setAndUnsetEntity() {
+    DatastoreService ds = DatastoreServiceFactory.getDatastoreService();
+
+    Set<String> CATS = new HashSet<>();
+        CATS.add(CAT_A);
+        CATS.add(CAT_B);
+        CATS.add(CAT_C);
+
+    Set<String> LINKS = new HashSet<>();
+        LINKS.add(LINK_A);
+        LINKS.add(LINK_B);
+
+    Marker expected_marker = new Marker(TITLE, DESCRIPT, LAT, LONG, LINKS, CATS, ID);
+        expected_marker.addFlagReport(FLAG_A);
+
+    assertEquals(0, ds.prepare(new Query("Marker")).countEntities());
+    Entity serialized_marker = expected_marker.toEntity();
+    ds.put(serialized_marker);
+    Marker deserialized_maker = new Marker(serialized_marker);
+    assertEquals(1, ds.prepare(new Query("Marker")).countEntities());
+
+    Assert.assertEquals(expected_marker.getUUID(), deserialized_maker.getUUID());
```

Comment:
I think you can drop the "Assert." since you imported it already?

Reply:
Yeah that's true! I was worried that it wouldn't work because it was static but it runs fine!